src/components/WaterCycle.vue

```diff
@@ -47,146 +46,244 @@
+const initialZoom = ref(zoom.value)
+
+
+// zooming with drag to pan when zoomed in
+const zoomStyle = computed(() => ({
+  transform: `translate(${pan.value.x}px, ${pan.value.y}px) scale(${zoom.value})`,
+  transformOrigin: transformOrigin.value,
+  transition: isDragging.value ? 'none' : 'transform 0.1s ease-out',
+  cursor: zoom.value > 1 ? (isDragging.value ? 'grabbing' : 'grab') : 'default',
+  userSelect: 'none',
+}))
+
+
+// center zoom on cursor
+const handleWheel = (e) => {
+  const rect = e.currentTarget.getBoundingClientRect()
+  const offsetX = e.clientX - rect.left
+  const offsetY = e.clientY - rect.top
+
+  const percentX = (offsetX / rect.width) * 100
+  const percentY = (offsetY / rect.height) * 100
+  transformOrigin.value = `${percentX}% ${percentY}%`
+
+  const delta = e.deltaY > 0 ? -0.1 : 0.1
+  zoom.value = Math.min(Math.max(zoom.value + delta, 1), 5)
+
+  // reset zoom center when at 1
+  if (zoom.value === 1) {
+    pan.value = clampPan(pan.value.x, pan.value.y)
+    transformOrigin.value = 'center center'
+  }
+}
+
+// mouse event handlers to allow pan effect
+const handleMouseDown = (e) => {
+  if (zoom.value <= 1) return
+  isDragging.value = true
+  dragStart.value = {
+    x: e.clientX - pan.value.x,
+    y: e.clientY - pan.value.y
+  }
+  // listeners for mouseup
+  window.addEventListener('mousemove', handleMouseMove)
+  window.addEventListener('mouseup', handleMouseUp)
+}
+
+const handleMouseMove = (e) => {
+  if (!isDragging.value) return
+  const rawX = e.clientX - dragStart.value.x
+  const rawY = e.clientY - dragStart.value.y
+  pan.value = clampPan(rawX, rawY)
+}
+
+const handleMouseUp = () => {
+  isDragging.value = false
+  window.removeEventListener('mousemove', handleMouseMove)
+  window.removeEventListener('mouseup', handleMouseUp)
+}
+
+// dont' scroll beyond the edges of the diagram...
+function clampPan(panX, panY) {
+  if (!zoomContainer.value || !imageWrapper.value) return { x: panX, y: panY }
+
+  const container = zoomContainer.value.getBoundingClientRect()
+  const image = imageWrapper.value.getBoundingClientRect()
+
+  const scale = zoom.value
+
+  const scaledWidth = image.width * scale
+  const scaledHeight = image.height * scale
+
+  const maxX = Math.max((scaledWidth - container.width) / 2, 0)
+  const maxY = Math.max((scaledHeight - container.height) / 2, 0)
+
+  return {
+    x: Math.max(Math.min(panX, maxX), -maxX),
+    y: Math.max(Math.min(panY, maxY), -maxY)
+  }
+}
+
+// on mobile calculate distance between fingers on screen
+function getTouchDistance(touches) {
+  const dx = touches[0].clientX - touches[1].clientX
+  const dy = touches[0].clientY - touches[1].clientY
+  return Math.sqrt(dx * dx + dy * dy)
+}
+
+// use touchpoints to drive zoom
+const handleTouchStart = (e) => {
+  if (e.touches.length === 2) {
+    initialPinchDistance.value = getTouchDistance(e.touches)
+    initialZoom.value = zoom.value
+  }
+}
+
+const handleTouchMove = (e) => {
```

Comment:
The pinch zoom is working beautifully smoothly on mobile, but it only seems to zoom to the center of the image. Also, once zoomed in, I can't pan.

RC: Improve centering of pinch zoom, add mobile panning functionalitly

Reply:
commits added, but I wasn't able to test via my local network so it's sort of a short in the dark